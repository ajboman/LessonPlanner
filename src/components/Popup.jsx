import { Modal, Button } from 'flowbite-react';

const Popup = ({ response, isVisible, onClose, onSave }) => {
    return (
        <Modal onClose={onClose} popup show={isVisible} size="md" color='gray'>
            <Modal.Header />
            <Modal.Body>
                <div className="text-center">
                    <h3 className="mb-5 text-lg font-normal text-black dark:text-gray-400">
                        Lesson Plan
                    </h3>
                    <p className='text-sm text-black whitespace-pre-wrap'>
                        {response}
                    </p>
                    <div className="flex justify-center gap-4 mt-5">
                        <Button
                            color="success"
                            onClick={onSave}
                        >
                            Save
                        </Button>
                        <Button
                            color="gray"
                            onClick={onClose}
                        >
                            Close
                        </Button>
                    </div>
                </div>
            </Modal.Body>
        </Modal>
    )
}

export default Popup;
