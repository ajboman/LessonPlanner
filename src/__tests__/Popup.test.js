import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Popup from '../components/Popup';

describe('Popup component', () => {
  const mockResponse = 'Test API response';
  const mockOnClose = jest.fn();
  const mockOnSave = jest.fn();

  test('renders correctly', () => {
    const { container } = render(
      <Popup response={mockResponse} isVisible={true} onClose={mockOnClose} onSave={mockOnSave} />
    );
    expect(container).toMatchSnapshot();
  });

  test('displays the response', () => {
    const { getByText } = render(
      <Popup response={mockResponse} isVisible={true} onClose={mockOnClose} onSave={mockOnSave} />
    );

    const responseElement = getByText(mockResponse);
    expect(responseElement).toBeInTheDocument();
  });

  test('calls onClose when Close button is clicked', () => {
    const { getByText } = render(
      <Popup response={mockResponse} isVisible={true} onClose={mockOnClose} onSave={mockOnSave} />
    );

    const closeButton = getByText('Close');
    fireEvent.click(closeButton);
    expect(mockOnClose).toHaveBeenCalled();
  });

  test('calls onSave when Save button is clicked', () => {
    const { getByText } = render(
      <Popup response={mockResponse} isVisible={true} onClose={mockOnClose} onSave={mockOnSave} />
    );

    const saveButton = getByText('Save');
    fireEvent.click(saveButton);
    expect(mockOnSave).toHaveBeenCalled();
  });
});
