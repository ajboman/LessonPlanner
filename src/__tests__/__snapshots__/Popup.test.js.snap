// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup component renders correctly 1`] = `
<div>
  <div
    class="fixed z-10 inset-0 overflow-y-auto"
  >
    <div
      class="flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0"
    >
      <div
        aria-hidden="true"
        class="fixed inset-0 transition-opacity"
      >
        <div
          class="absolute inset-0 bg-gray-500 opacity-75"
        />
      </div>
      <span
        aria-hidden="true"
        class="hidden sm:inline-block sm:align-middle sm:h-screen"
      >
        â€‹
      </span>
      <div
        class="inline-block align-bottom bg-white rounded-lg text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full"
      >
        <div
          class="bg-white px-4 pt-5 pb-4 sm:p-6 sm:pb-4"
        >
          <div
            class="sm:flex sm:items-start"
          >
            <div
              class="mt-3 text-center sm:mt-0 sm:ml-4 sm:text-left"
            >
              <h3
                class="text-lg leading-6 font-medium text-gray-900"
              >
                API Response
              </h3>
              <div
                class="mt-2 whitespace-pre-wrap"
              >
                <p
                  class="text-sm text-gray-500"
                >
                  Test API response
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse"
        >
          <button
            class="w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-blue-600 text-base font-medium text-white hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 sm:ml-3 sm:w-auto sm:text-sm"
            type="button"
          >
            Save
          </button>
          <button
            class="mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
